src/main/java/net/snowflake/client/core/SFBaseSession.java

```diff
@@ -333,15 +339,28 @@ public HttpClientSettingsKey getHttpClientKey() throws SnowflakeSQLException {
       String nonProxyHosts =
           (String) connectionPropertiesMap.get(SFSessionProperty.NON_PROXY_HOSTS);
       String proxyProtocol = (String) connectionPropertiesMap.get(SFSessionProperty.PROXY_PROTOCOL);
-      ocspAndProxyKey =
-          new HttpClientSettingsKey(
-              getOCSPMode(),
-              proxyHost,
-              proxyPort,
-              nonProxyHosts,
-              proxyUser,
-              proxyPassword,
-              proxyProtocol);
+      if (userAgentSuffix.isEmpty()) {
```

Comment:
As such this can be optimized as the constructor will set empty userAgentSuffix.